File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/context/XenonBeanInject.java
package com.abubusoft.xenon.context;

import com.abubusoft.xenon.context.XenonBeanType;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;


/**
 * <p>Effettua l'inject del campo con l'entità argon specificata.</p> 
 * @author Francesco Benincasa
 *
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(value=ElementType.FIELD)
public @interface XenonBeanInject {

	/**
	 * <p>
	 * Indica il tipo di injection da inserire.
	 * </p>
	 * 
	 * @return
	 */
	XenonBeanType value();
	
}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/context/XenonBeanInject.kt
package com.abubusoft.xenon.context

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 *
 * Effettua l'inject del campo con l'entità argon specificata.
 * @author Francesco Benincasa
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(AnnotationTarget.FIELD)
annotation class XenonBeanInject(
    /**
     *
     *
     * Indica il tipo di injection da inserire.
     *
     *
     * @return
     */
    val value: XenonBeanType
)